### PR TITLE
Golem AI

### PR DESCRIPTION
**Pull request by Caedus-Ustrina**
Fixed collision on village and forge mountain
Golem abstract class added
Hostile Golem added

- Hostile Golem chases player when in range
- Hostile Golem marches left when no target in range
- Hostile Golem will jump up surfaces that block its path (pretty sure it can multiple jump atm though)

### CODE REVIEW

File: Entities/Golem/EnemyGolemAI.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+			this._Velocity.X = (int)this.Bearing * Speed;
+
+			this.Velocity = this._Velocity;
+
+			MoveAndSlide();
+		}
+
+		private bool SearchForPlayerEntity()
+		{
```
Comment (Rambolink94) [inline Entities/Golem/EnemyGolemAI.cs:45]: Move this call to `_Ready()`

File: Entities/Golem/GolemAI.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		protected Node2D Target = new Node2D();
+
+		protected abstract Bearing ObjectiveBearing { get; }
+
+		// Get the gravity from the project settings to be synced with RigidBody nodes.
+		public float gravity = ProjectSettings.GetSetting("physics/2d/default_gravity").AsSingle();
+
+		protected bool FindWalls()
+		{
+			Area2D wallDetector = GetNode<Area2D>("Wall Detector");
```
Comment (Rambolink94) [inline Entities/Golem/GolemAI.cs]: Move this call to `_Ready()` method

**Review by Rambolink94 — CHANGES_REQUESTED** [review]

File: Entities/Golem/EnemyGolemAI.cs (hunk 1 above)
Comment (Rambolink94) [reply Entities/Golem/EnemyGolemAI.cs:45]: It looks like it didn't correctly highlight what I mean to comment on. I was referening the `GetNode` call in the method.

Comment (Caedus-Ustrina) [reply Entities/Golem/EnemyGolemAI.cs:45]: Diiid it

File: Entities/Golem/GolemAI.cs (hunk 2 above)
Comment (Caedus-Ustrina) [reply Entities/Golem/GolemAI.cs]: Diiid it

**Review by Rambolink94 — APPROVED** [review]

**Review by Rambolink94 — APPROVED** [review]

**Pull request merged**